src/App.tsx

```diff
@@ -1,24 +1,166 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import { UserWarning } from './components/UserWarning';
+import {
+  getTodosFromServer,
+  addTodoOnServer,
+  deleteTodoOnServer,
+} from './api/todos';
 
-const USER_ID = 0;
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { ErrorNotification } from './components/ErrorNotification';
+
+import { TodoCompletionType } from './types/TodoCompletionType';
+import { ErrorType } from './types/ErrorType';
+import { TodoRich } from './types/TodoRich';
+
+import { filterTodos, getActiveTodosCount } from './common/helpers';
+import { Todo } from './types/Todo';
+
+const USER_ID = 6955;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<TodoRich[]>([]);
+  const [error, setError] = useState<ErrorType>(ErrorType.None);
+  const [selectedFilterOption, setSelectedFilterOption]
+    = useState(TodoCompletionType.All);
+
+  const [tempTodo, setTempTodo] = useState<TodoRich | null>(null);
+
+  const setLoadingTodo = (todoId: number, isLoading: boolean): void => (
+    setTodos(prevTodos => (
+      prevTodos.map(todo => {
+        if (todo.id !== todoId) {
+          return todo;
+        }
+
+        return {
+          ...todo,
+          isLoading,
+        };
+      })
+    ))
+  );
+
+  const addTodo = (newTodo: Todo): void => {
+    setTodos((prevTodos) => (
+      [
+        ...prevTodos,
+        {
+          ...newTodo,
+          isLoading: false,
+        },
+      ]
+    ));
+  };
+
+  const deleteTodo = (todoId: number): void => {
+    setTodos(prevTodos => (
+      prevTodos.filter(todo => todo.id !== todoId)
+    ));
+  };
+
+  const handleTodoAdd = async (todoTitle: string): Promise<void> => {
+    if (todoTitle === '') {
+      setError(ErrorType.EmptyTitle);
+
+      return;
+    }
+
+    const newTodo: TodoRich = {
+      id: 0,
+      title: todoTitle,
+      completed: false,
+      userId: USER_ID,
+      isLoading: true,
+    };
```

Comment:
I don't think it's a good idea to add the isLoading property to the Todo object type. Instead, it would be better to manage the component's state and props to control the loading logic. Modifying or creating new types just to add props can make the code more complicated and harder to maintain.